helios-system-tests/src/main/java/com/spotify/helios/system/AgentArgsBuilder.java

```diff
@@ -0,0 +1,170 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.helios.system;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Range;
+
+import com.spotify.docker.client.DockerHost;
+
+import org.apache.commons.lang.StringUtils;
+
+import java.util.List;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+
+/**
+ * Class that helps build program arguments for {@link com.spotify.helios.agent.AgentMain}
+ */
+public class AgentArgsBuilder {
+
+  private int verbosity = 4;
+  private boolean noLogSetup = true;
+  private boolean noHttp = true;
+  private String httpEndpoint;
+  private String adminEndpoint;
+  private String name;
+  private DockerHost dockerHost;
+  private String zkConnectString;
+  private int zkSessionTimeout = 100;
+  private int zkConnectionTimeout = 100;
+  private boolean zkEnableAcls = true;
+  private String zkAclMasterUser;
+  private String zkAclMasterDigest;
+  private String zkAclAgentUser;
+  private String zkAclAgentPassword;
+  private String stateDir;
+  private String domain = "";
+  private Range<Integer> dockerPortRange;
+
+  private AgentArgsBuilder() {
+  }
+
+  public AgentArgsBuilder name(final String name) {
+    this.name = name;
+    return this;
+  }
+
+  public AgentArgsBuilder dockerHost(final DockerHost dockerHost) {
+    this.dockerHost = dockerHost;
+    return this;
+  }
+
+  public AgentArgsBuilder zkAclMasterUser(final String zkAclMasterUser) {
+    this.zkAclMasterUser = zkAclMasterUser;
+    return this;
+  }
+
+  public AgentArgsBuilder zkAclMasterDigest(final String zkAclMasterDigest) {
+    this.zkAclMasterDigest = zkAclMasterDigest;
+    return this;
+  }
+
+  public AgentArgsBuilder zkAclAgentUser(final String zkAclAgentUser) {
+    this.zkAclAgentUser = zkAclAgentUser;
+    return this;
+  }
+
+  public AgentArgsBuilder zkAclAgentPassword(final String zkAclAgentPassword) {
+    this.zkAclAgentPassword = zkAclAgentPassword;
+    return this;
+  }
+
+  public AgentArgsBuilder zkConnectString(final String zkConnectString) {
+    this.zkConnectString = zkConnectString;
+    return this;
+  }
+
+  public AgentArgsBuilder stateDir(final String stateDir) {
+    this.stateDir = stateDir;
+    return this;
+  }
+
+  public AgentArgsBuilder dockerPortRange(final Range<Integer> dockerPortRange) {
+    this.dockerPortRange = dockerPortRange;
+    return this;
+  }
+
+  public AgentArgsBuilder noHttp(final boolean noHttp) {
+    this.noHttp = noHttp;
+    return this;
+  }
+
+  public AgentArgsBuilder httpEndpoint(final String httpEndpoint) {
+    this.httpEndpoint = httpEndpoint;
+    return this;
+  }
+
+  public AgentArgsBuilder adminEndpoint(final String adminEndpoint) {
+    this.adminEndpoint = adminEndpoint;
+    return this;
+  }
+
+  public List<String> build() {
```

Comment:
Should I add some invariants here like `checkNotNull(dockerHost)`? And if so which ones?